OZON.ru.js

```diff
@@ -9,596 +9,130 @@
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText;
-		switch(nameText) {
-		case "От производителя":
-			valueText =
-			valueDOMNode.innerHTML.replace(/^[\s\S]*ANNOTATION\] -->/, "");
-			//Zotero.debug("valueText -> " + valueText);
-			valueText = Zotero.Utilities.cleanTags(valueText);
-			result["abstractNote"] = valueText;
-			break;
-		case "От OZON.ru":
-			valueText =
-			valueDOMNode.textContent.replace(/^[\s\S]*От OZON.ru/, "");
-			valueText = CleanText(valueText);
-			result["edition"] = GetEdition(valueText);
-			break;
-		}
+function createCreator(type, name) {
+	var name = splitFullName(name);
+	return {
+		creatorType: type,
+		firstName: name.first,
+		lastName: name.last
 	}
-	return result;
 }
-
-function doWeb(doc, url) {
-	var pdObject = new Object();
-	//pdObject["url"] = url;
-
-	// For the reason I do not understand, OZON.ru may alter its layout
-	// even within the same browser and screen resolution. I only noticed
-	// two layout types so far. Therefore the xpath expressions are
-	// extended with 'or' operator to accommodate both layouts types.
-
-	var pdXPath = '//div[@class="bContentBlock"]/h1|//div[@class="l h1"]/h1';
-	var pdXPathRes = doc.evaluate(
-	pdXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject["title"] = CleanText(pdXPathRes.iterateNext().textContent);
-	pdXPath = '//div[@id="js_basic_properties"]/p|//div[@class="product-detail"]/p';
-	pdXPathRes = doc.evaluate(
-	pdXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject = MergeObjects(pdObject, ParseProductDetail(pdXPathRes));
-
-	var nameXPath = '//div[@class="bTechDescription"]/div/div[2]/span|//div[@class="techDescription"]/div/div[2]/span';
-	var valueXPath = '//div[@class="bTechDescription"]/div/div[3]/span|//div[@class="techDescription"]/div/div[3]/span';
-	var nameXPathRes = doc.evaluate(
-	nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-	var valueXPathRes = doc.evaluate(
-	valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject = MergeObjects(
-	pdObject, ParseTechDesc(nameXPathRes, valueXPathRes));
-
-	nameXPath = '//div[@id="detail_description"]/div/table/tbody/tr/td/h3|//div[@id="detail_description"]/table/tbody/tr/td/h3';
-	valueXPath = '//div[@id="detail_description"]/div/table/tbody/tr/td|//div[@id="detail_description"]/table/tbody/tr/td';
-	nameXPathRes = doc.evaluate(
-	nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-	valueXPathRes = doc.evaluate(
-	valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-	pdObject = MergeObjects(
-	pdObject, ParseDetailDesc(nameXPathRes, valueXPathRes));
-
-	var newItem = new Zotero.Item("book");
-	for (var field in pdObject) newItem[field] = pdObject[field];
-
-	var linkurl = url;
-	newItem.attachments = [{
-		"url": linkurl,
-		"title": "OZON.ru Link",
-		"mimeType": "text/html",
-		"snapshot": false
-	}];
-
-	newItem.complete();
-}
-/** BEGIN TEST CASES **/
-var testCases = [
```

Comment:
Would it be possible to keep tests? I.e. are the URLs still the same? If no, then can we fix the URLs to refer to the same articles?

Also, looks like you're not using [Scaffold](https://www.zotero.org/support/dev/translators/scaffold). Please do. It will make your life easier (though you need Firefox)